pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -948,6 +959,20 @@ func (p *Pod) ReclaimablePods() ([]kueue.ReclaimablePod, error) {
 	return result, nil
 }
 
+func (p *Pod) finalizeInactivePods(ctx context.Context, c client.Client) error {
```

Comment:
Similarly here, I don't have a better name to propose, but the "active/inactive" makes it slightly confusing. Maybe "finalizeNonRunnablePods"?